### PR TITLE
Rewrote to es6 classes. Added prop-types for rn >= 0.44.

### PR DESCRIPTION
**Pull request by bintoll**
Added prop-types dependency and fix this in code. Rewrote two component s in es6 classes.

### CODE REVIEW

**Comment by justim** [comment]
Your `package.json` has a syntax error, can you please fix this?

**Comment by bintoll** [comment]
Yes, done.

**Comment by justim** [comment]
Thanks! Lightbox is now at least usable again with your PR 👍 

For anyone else wanting to use this PR:

```js
{
  // ...
  "dependencies": {
    // ...
    "react-native-lightbox": "github:oblador/react-native-lightbox#ade502ae2d25ff0031e0d8dce703606ce280f822",
  }
}
```

**Comment by oblador** [comment]
This is great, thanks! I'll review and test as soon as i get the chance. Would you care helping me maintain the lib?

**Comment by bintoll** [comment]
If you need a help - i can, of course.

File: Lightbox.js [hunk 1]
```diff
@@ -3,49 +3,16 @@
...
-      onClose: () => {},
-    };
-  },
-
-  getInitialState: function() {
-    return {
+class Lightbox extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (oblador) [inline Lightbox.js]: This can be defined directly as an instance variable.

File: Lightbox.js [hunk 2]
```diff
@@ -152,6 +117,27 @@ var Lightbox = React.createClass({
       </View>
     );
   }
-});
+}
+
+Lightbox.propTypes = {
```
Comment (oblador) [inline Lightbox.js]: I prefer if these are defined as static class variables instead :-)

File: Lightbox.js [hunk 3]
```diff
@@ -152,6 +117,27 @@ var Lightbox = React.createClass({
...
+  onOpen:          PropTypes.func,
+  onClose:         PropTypes.func,
+  springConfig:    PropTypes.shape({
+    tension:       PropTypes.number,
+    friction:      PropTypes.number,
+  }),
+  swipeToDismiss:  PropTypes.bool,
+},
+
+Lightbox.defaultProps = {
```
Comment (oblador) [inline Lightbox.js]: Same

File: LightboxOverlay.js [hunk 4]
```diff
@@ -3,50 +3,19 @@
...
-  StyleSheet,
-  Text,
-  TouchableOpacity,
-  View,
-} = require('react-native');
+import React, { Component } from 'react';
+import PropTypes from 'prop-types'
+import { Animated, Dimensions, Modal, PanResponder, Platform, StatusBar, StyleSheet, Text, TouchableOpacity, View } from 'react-native'
 
 var WINDOW_HEIGHT = Dimensions.get('window').height;
```
Comment (oblador) [inline LightboxOverlay.js]: These should be `const`

**Comment by bintoll** [comment]
Okay, i'll rewrite it tomorrow, it's not a problem. First time the main propose was just to move to es6 classes and not to beautify code) But okay, i also prefer good looking code)

**Comment by oblador** [comment]
OK, merging this and fixing the syntax separately instead.

**Comment by oblador** [comment]
Released in [0.7.0](https://github.com/oblador/react-native-lightbox/releases/tag/v0.7.0)

**Pull request merged**